stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/files/ChatInfoSharedFilesViewHolders.kt

```diff
@@ -0,0 +1,54 @@
+package io.getstream.chat.ui.sample.feature.chat.info.shared.files
+
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.utils.MediaStringUtil
+import io.getstream.chat.android.core.internal.InternalStreamChatApi
+import io.getstream.chat.android.ui.utils.UiUtils
+import io.getstream.chat.ui.sample.databinding.ChatInfoSharedFileDateDividerBinding
+import io.getstream.chat.ui.sample.databinding.ChatInfoSharedFileItemBinding
+import io.getstream.chat.ui.sample.feature.chat.info.shared.SharedAttachment
+import java.text.DateFormat
+
+abstract class BaseViewHolder<T : SharedAttachment>(
+    itemView: View
+) : RecyclerView.ViewHolder(itemView) {
+
+    /**
+     * Workaround to allow a downcast of the SharedAttachment to T
+     */
+    @Suppress("UNCHECKED_CAST")
+    internal fun bindListItem(item: SharedAttachment) = bind(item as T)
+
+    protected abstract fun bind(item: T)
+}
+
+class ChatInfoSharedFileViewHolder(
+    private val binding: ChatInfoSharedFileItemBinding,
+    attachmentClickListener: ChatInfoSharedFilesAdapter.AttachmentClickListener?
+) : BaseViewHolder<SharedAttachment.AttachmentItem>(binding.root) {
+
+    private lateinit var attachmentItem: SharedAttachment.AttachmentItem
+
+    init {
+        binding.fileItemContainer.setOnClickListener { attachmentClickListener?.onClick(attachmentItem) }
+    }
+
+    @InternalStreamChatApi
```

Comment:
This should not happen in the sample outside of the component browser, anything that's needed to build the sample should be public in the SDK or implemented within the sample if necessary